lib/ecto/changeset.ex

```diff
@@ -568,7 +570,10 @@ defmodule Ecto.Changeset do
             end
 
           :error ->
-            :invalid
+            {:invalid, message: "is invalid"}
+
+          {:error, custom_errors} ->
```

Comment:
I would add a guard, just in case:

```suggestion
          {:error, custom_errors} when is_list(custom_errors) ->
```